pkg/mdformatter/mdformatter.go

```diff
@@ -109,11 +110,31 @@ func (FormatFrontMatter) TransformFrontMatter(_ SourceContext, frontMatter map[s
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(keys)))
 
-	b := bytes.NewBuffer([]byte("---\n"))
+	b := bytes.NewBuffer([]byte("---"))
 	for _, k := range keys {
-		_, _ = fmt.Fprintf(b, "%v: %v\n", k, frontMatter[k])
+		// check if frontMatter[k] is a map
+		frontMatterStr := fmt.Sprintf("%s", frontMatter[k])
```

Comment:
This is nice tricky, but wonder if we could add some type safety (: Would ` _, isMap := frontMatter[k].(map[string]interface{})` work? 🤔

Reply:
Will check it out!

Reply:
Looks like it worked! Do we need `frontMatterStr` then? 🤗